src/conv/invokers/mlir_impl_gemm.cpp

```diff
@@ -135,33 +164,67 @@ InvokerFactory MakeMlirFwdInvokerFactory(const ConvolutionContext& ctx)
                 primitive_parameters.CastTo<conv::DataInvokeParams>();
             const auto& tensors = forward_invoke_params.tensors;
 
-            void* filter = nullptr;
-            void* input  = nullptr;
-            void* output = nullptr;
-#if MIOPEN_BACKEND_OPENCL
-            clGetMemObjectInfo(tensors.w, CL_MEM_HOST_PTR, sizeof(filter), &filter, nullptr);
-            clGetMemObjectInfo(tensors.in, CL_MEM_HOST_PTR, sizeof(input), &input, nullptr);
-            clGetMemObjectInfo(tensors.out, CL_MEM_HOST_PTR, sizeof(output), &output, nullptr);
-#elif MIOPEN_BACKEND_HIP
-            // NOLINTNEXTLINE (cppcoreguidelines-pro-type-const-cast)
-            filter = const_cast<void*>(tensors.w);
-            // NOLINTNEXTLINE (cppcoreguidelines-pro-type-const-cast)
-            input = const_cast<void*>(tensors.in);
-            // NOLINTNEXTLINE (cppcoreguidelines-pro-type-const-cast)
-            output = const_cast<void*>(tensors.out);
-#endif
+            setMlirConvArgsPtr(tensors, args);
+            handle.Run(kernels[0])(args);
+        };
+    };
+}
+
+InvokerFactory MakeMlirBwdInvokerFactory(const ConvolutionContext& ctx)
+{
+    assert(ctx.direction.IsBackwardData());
+
+    std::vector<size_t> in_dims, in_strides;
+    std::vector<size_t> weights_dims, weights_strides;
+    std::vector<size_t> out_dims, out_strides;
+    permuteDimStridesAllDir(ctx.conv_problem,
+                            in_dims,
+                            in_strides,
+                            weights_dims,
+                            weights_strides,
+                            out_dims,
+                            out_strides);
+    MlirConvArgs args =
+        makeMlirConvArgs(in_dims, in_strides, weights_dims, weights_strides, out_dims, out_strides);
+
+    const auto& conv = ctx.conv_problem.GetConv();
+    const auto& wDesc = ctx.conv_problem.GetWeights();
+    bool is_1x1_s1    = false;
+    if(miopen::all_of(conv.GetConvPads(), [](auto v) { return v == 0; }) &&
+       miopen::all_of(conv.GetConvStrides(), [](auto v) { return v == 1; }))
+    {
+        if(wDesc.GetLengths()[2] == 1 && wDesc.GetLengths()[3] == 1)
+        { // filter = 1
+            if(wDesc.GetSize() == 4 || (wDesc.GetSize() == 5 && wDesc.GetLengths()[4] == 1))
+            {
+                is_1x1_s1 = true;
+            }
+        }
+    }
```

Comment:
[Recommendation]
```C++
const auto isFilter1Padding0Stride1 = [](){
    const auto wei_spatial = boost::adaptors::slice(wDesc.GetLengths(), 2, 2 + conv.GetSpatialDimension());
    return  miopen::all_of(wei_spatial, [](auto v) { return v == 1; }) &&
       miopen::all_of(conv.GetConvPads(), [](auto v) { return v == 0; }) &&
       miopen::all_of(conv.GetConvStrides(), [](auto v) { return v == 1; });
}();
```

Reply:
Nice suggestion! Does make it much cleaner than nested if conditions. Addressed in 30aca46

Reply:
And now we have `const` object!